pkg/kubectl/resource/visitor.go

```diff
@@ -461,32 +465,33 @@ func (v *StreamVisitor) Visit(fn VisitorFunc) error {
 			if err == io.EOF {
 				return nil
 			}
-			return err
+			return fn(nil, err)
 		}
 		ext.RawJSON = bytes.TrimSpace(ext.RawJSON)
 		if len(ext.RawJSON) == 0 || bytes.Equal(ext.RawJSON, []byte("null")) {
 			continue
 		}
 		if err := ValidateSchema(ext.RawJSON, v.Schema); err != nil {
-			return err
+			return fn(nil, err)
 		}
 		info, err := v.InfoForData(ext.RawJSON, v.Source)
 		if err != nil {
-			if v.IgnoreErrors {
```

Comment:
Before the visitor accepted an error IgnoreErrors was required.  With it
being part of Visit we can remove it (but higher levels need to properly
handle it).  The original goal was to allow as much expansion as possible
as late as possible.

On Aug 16, 2015, at 3:28 AM, Chao Xu <notifications@github.com> wrote:

In pkg/kubectl/resource/visitor.go
<https://github.com/kubernetes/kubernetes/pull/9787#discussion_r37142970>:

>  		}
>  		info, err := v.InfoForData(ext.RawJSON, v.Source)
>  		if err != nil {
> -			if v.IgnoreErrors {

IgnoreErrors is not used elsewhere. So we can remove this field.

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/9787/files#r37142970>.